controllers/apidefinition_controller.go

```diff
@@ -132,29 +132,54 @@ func (r *ApiDefinitionReconciler) Reconcile(ctx context.Context, req ctrl.Reques
 		// we support only one certificate secret name for mvp
 		if len(desired.Spec.CertificateSecretNames) != 0 {
 			certName := desired.Spec.CertificateSecretNames[0]
-			tykCertID, err := r.checkSecretAndUpload(ctx, certName, namespacedName, log, &env)
+			tykCertID, err := r.checkSecretAndUpload(ctx, certName, namespacedName.Namespace, log, &env, "")
 			if err != nil {
 				return err
 			}
 
 			upstreamRequestStruct.Spec.Certificates = []string{tykCertID}
 		}
 
+		// Check Pinned Public keys
+		if len(desired.Spec.PinnedPublicKeysSecretNames) != 0 {
+			if upstreamRequestStruct.Spec.PinnedPublicKeys == nil {
+				upstreamRequestStruct.Spec.PinnedPublicKeys = map[string]string{}
+			}
+
+			for domain, secretName := range desired.Spec.PinnedPublicKeysSecretNames {
+				// Set the namespace for referenced secret to the current namespace where ApiDefinition lives.
+				tykCertID, err := r.checkSecretAndUpload(ctx, secretName, req.Namespace, log, &env, "public-key")
```

Comment:
if `public-key` is static, then it should be a constant. But I don't know if it is a good idea to be explicit about it - e.g. a public key generated by cert-manager might be called something like `tls.crt`. Is it worth considering how we can be a bit more interoperable with standard tooling?